src/providers/class-pantheon-provider.php

```diff
@@ -0,0 +1,473 @@
+<?php
+/**
+ * Pantheon_Provider class file
+ *
+ * @package wp-page-cache-control
+ */
+
+namespace Alley\WP\WP_Page_Cache_Control\Providers;
+
+use Alley\WP\WP_Page_Cache_Control\Header;
+use InvalidArgumentException;
+use Mantle\Support\Str;
+use WP_Post;
+use WP_Term;
+
+use function Mantle\Support\Helpers\collect;
+
+/**
+ * Pantheon Advanced Page Cache Provider
+ *
+ * @link https://docs.pantheon.io/cookies#cache-varying-cookies
+ */
+class Pantheon_Provider implements Provider {
+	use Concerns\Interacts_With_IP_Addresses,
+		Concerns\Manages_Cookies;
+
+	/**
+	 * Cookie name for the no-cache cookie.
+	 *
+	 * @var string
+	 */
+	const COOKIE_NO_CACHE = 'NO_CACHE';
+
+	/**
+	 * Cookie prefix for cache groups.
+	 *
+	 * @var string
+	 */
+	const COOKIE_SEGMENT_PREFIX = 'STYXKEY-';
+
+	/**
+	 * Determine if the user is not being cached (bypassing the page cache with
+	 * a cookie).
+	 *
+	 * @var bool
+	 */
+	protected bool $is_user_no_cache = false;
+
+	/**
+	 * Storage of cache groups.
+	 *
+	 * @var array<string, string>
+	 */
+	protected array $groups = [];
+
+	/**
+	 * Flag if we should update the group cookies.
+	 *
+	 * @var bool
+	 */
+	protected bool $should_update_group_cookies = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @throws InvalidArgumentException If the Vary_Cache class does not exist.
```

Comment:
`Vary_Cache` is VIP and this is the Pantheon provider, so I think this docblock comment should be removed.